service/controller/v14/resource/endpoint/delete.go

```diff
@@ -20,6 +20,8 @@ func (r *Resource) ApplyDeleteChange(ctx context.Context, obj, deleteState inter
 		return nil // Nothing to do.
 	}
 
+	// TODO this looks very wrong. When the endpoint is not empty on a delete
+	// event, we want to delete it usually. The code below says we don't.
 	if !isEmptyEndpoint(*k8sEndpoint) {
 		return nil
 	}
```

Comment:
Just something I discovered. Need opinions here. This looks pretty wrong.

Reply:
Ok I see. The `endpoint` resource is reconciled by watching pods. Pods get deleted at times. We do not want to delete endpoints only because one pod is gone. So yes, this makes sense. I will add some comments to explain that. Thank you.